core/src/main/java/io/syndesis/core/Tokens.java

```diff
@@ -38,13 +49,14 @@ public static String getAuthenticationToken() {
         Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
         if (authentication == null) {
             throw new IllegalStateException("Cannot set authorization header because there is no authenticated principal");
-        } else if (!KeycloakAuthenticationToken.class.isAssignableFrom(authentication.getClass())) {
+        }
```

Comment:
I think this method should be split into two methods as they serve completely different use cases:

* `getAuthenticationTokenForController` is called by the integration controllers which do the status updates (e.g. updating the files in git). This method looks up the token from a ThreadLocal (the first four lines in this method here). The caller of this method will never have an SecurityContextHolder from a web request set, so when the caller calls this method and there is no such token, it should error immediately (leading to a clearer errror message)

* `getAuthenticationTokenFromSecurityContext` is called by everyone in course of a web request handler chain. It should never lookup in the (controller reserved) thread local variable. 

Actually, I would put these methods even on different classes (the controller related more closely to the controllers themselves)